### PR TITLE
Fix background processing of Enterprise Fee Summary report

### LINKED ISSUE #10395: Enterprise Fee Summary report fails on uk-staging with with background reports feature
## Description

@drummer83 found this bug testing the new background reports:

* https://github.com/openfoodfoundation/openfoodnetwork/pull/10340#issuecomment-1421503351

## Expected Behavior

The report renders.

## Actual Behaviour

Error 500.

## Steps to Reproduce

1. Load uk-staging data
2. Login as super admin
3. Visit /admin/reports/enterprise_fee_summary
4. Try to show a report

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

## Error in OpenFoodNetwork UK

**ActionView::Template::Error** in **admin/reports#show**
undefined method `new' for nil:NilClass

[View on Bugsnag](https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/63e2997bc5538c000902c446?event_id=63e2c25800add59550ae0000&i=gh&m=ci)

## Stacktrace

    lib/reporting/reports/enterprise_fee_summary/summarizer.rb:27 - representation
    lib/reporting/reports/enterprise_fee_summary/summarizer.rb:9 - fee_type
    lib/reporting/reports/enterprise_fee_summary/base.rb:75 - public_send
    lib/reporting/reports/enterprise_fee_summary/base.rb:75 - block (3 levels) in enterprise_fee_type_total_list
    lib/reporting/reports/enterprise_fee_summary/base.rb:74 - each
    lib/reporting/reports/enterprise_fee_summary/base.rb:74 - block (2 levels) in enterprise_fee_type_total_list
    lib/reporting/reports/enterprise_fee_summary/base.rb:73 - block in enterprise_fee_type_total_list
    lib/reporting/reports/enterprise_fee_summary/base.rb:70 - map
    lib/reporting/reports/enterprise_fee_summary/base.rb:70 - enterprise_fee_type_total_list
    lib/reporting/reports/enterprise_fee_summary/base.rb:35 - query_result
    lib/reporting/report_rows_builder.rb:41 - computed_data
    lib/reporting/report_rows_builder.rb:15 - grouped_data
    lib/reporting/report_template.rb:11 - grouped_data
    app/views/admin/reports/_table.html.haml:9 - _app_views_admin_reports__table_html_haml___417132116801927730_138660
    lib/reporting/report_renderer.rb:50 - to_html
    lib/reporting/report_renderer.rb:46 - public_send
    lib/reporting/report_renderer.rb:46 - render_as
    lib/reporting/report_template.rb:8 - render_as
    app/jobs/report_job.rb:7 - perform
    app/services/job_processor.rb:14 - block (2 levels) in perform_forked
    app/services/job_processor.rb:13 - block in perform_forked
    app/services/job_processor.rb:11 - perform_forked
    app/controllers/admin/reports_controller.rb:59 - render_report_as
    app/controllers/admin/reports_controller.rb:48 - assign_view_data
    app/controllers/admin/reports_controller.rb:38 - show_report
    app/controllers/admin/reports_controller.rb:27 - show
    lib/session_cookie_upgrader.rb:19 - call

[View full stacktrace](https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/63e2997bc5538c000902c446?event_id=63e2c25800add59550ae0000&i=gh&m=ci)

*Created by Maikel via Bugsnag*

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

- Closes #10395 

#### What should we test?

- Login as super admin on uk-staging
- Activate background reports feature.
- Visit /admin/reports/enterprise_fee_summary
- Try to show a report

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by jibees — APPROVED** [review]
All clear, thanks!

Some linter errors though.

**Comment by mkllnk** [comment]
> Some linter errors though.

Ah, of course. Should be fixed now. :crossed_fingers:

**Review by jibees — APPROVED** [review]
Thanks!

**Comment by sigmundpetersen** [comment]
I guess this should rather go to Test ready than to In Dev? @jibees

**Comment by jibees** [comment]
> I guess this should rather go to Test ready than to In Dev? @jibees

Yes, I missed the right state ;) Thanks for your amazing watchfulness!

**Comment by drummer83** [comment]
Hi @mkllnk,

Sorry, unless I did something wrong, there is still the error 500 for super admin user.

A second Bugsnag was triggered at the same time:
https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/63e2b01a5f6f3500084cc414?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage]=staging
This is the one you already linked to the issue: 
https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/63e2997bc5538c000902c446?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage]=staging

Same on staging AU:
https://app.bugsnag.com/yaycode/openfoodnetwork-aus/errors/63eb9c1317beb00008cc888c?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage]=staging
https://app.bugsnag.com/yaycode/openfoodnetwork-aus/errors/63e28a701dfcc70008369e26?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage]=staging

Moving back to In Dev.

**Comment by mkllnk** [comment]
Thanks for testing. I actually see an error now when the feature toggle is off.

**Comment by mkllnk** [comment]
Hm, same on staging with master. We got a different problem here in master, nothing to do with background reports.

**Comment by RachL** [comment]
for testing purposes: this error could be on other report. From this discussion [here](https://openfoodnetwork.slack.com/archives/CN366FPS9/p1676412330954369?thread_ts=1674137746.945829&cid=CN366FPS9) it seems I was experiencing this issue on the tax report as well, so might not be linked to the enterprise fee summary report only?

File: lib/reporting/reports/enterprise_fee_summary/scope.rb [hunk 1]
```diff
@@ -317,7 +317,8 @@ def include_outgoing_exchange_details
         def filter_by_date(params)
           filter_scope("spree_orders.completed_at >= ?", params.completed_at_gt) \
             if params.completed_at_gt.present?
-          filter_scope("spree_orders.completed_at <= ?", params.completed_at_lt) if params.completed_at_lt.present?
+          filter_scope("spree_orders.completed_at <= ?", params.completed_at_lt) \
+            if params.completed_at_lt.present?
```
Comment (dacook) [inline lib/reporting/reports/enterprise_fee_summary/scope.rb:321]: I think I would have chosen an `if` block here, but rubocop might disagree!

**Review by dacook — APPROVED** [review]
Good pickup, and thanks for cleaning it up!

**Comment by dacook** [comment]
I think this is ready for test, however can't be tested until #10442  is merged. Not sure where to put it so I will move it to the bottom of the `Test ready" queue..

**Comment by drummer83** [comment]
Hi @mkllnk,

Thanks for improving the report generation. We only have 3 situations left, where I ended up with Internal Server Errors. You can find links to Bugsnag below. I did not re-test normal admin users, because they didn't make problems last time already.

| Report type | Background reports activated | **admin** user (**w/o** date restriction) | **admin** user (**with** date restriction) | **super admin** user (**w/o** date restriction) | **super admin** user (**with** date restriction) | 
| - | - | - | - | - | - |
| Orders And Distributors | yes | not tested | not tested | :x: Timeout 504 | :heavy_check_mark: |
| Bulk Co-op Supplier Report | yes | not tested | not tested | :x: Timeout 504 | :heavy_check_mark: |
| Payments By Type | yes | not tested | not tested | :x: Internal Server Error 500 (21:41) | :x: Internal Server Error 500 (21:44) |
| Order Cycle Supplier Totals | yes | not tested | not tested | :x: Internal Server Error (21:48) | :heavy_check_mark: |
| Mailing List | yes | not tested | not tested | :heavy_check_mark: | n.a. |
| All products | yes | not tested | not tested | :x: Timeout 504 | n.a. |
| Users & Enterprises | yes | not tested | not tested | :heavy_check_mark: | n.a. |
| Enterprise Fee Summary | yes | not tested | not tested | :heavy_check_mark: | :heavy_check_mark: |
| Payment Methods Report | yes | not tested | not tested | :x: Timeout 504 | :heavy_check_mark: |
| Tax Types | yes | not tested | not tested | :x: Timeout 504 | :heavy_check_mark: |
| Xero Summary | yes | not tested | not tested | :x: Timeout 504 | :heavy_check_mark: |
| Pack By Customer | yes | not tested | not tested | :x: Timeout 504 | :heavy_check_mark: |
| Revenues By Hub | yes | not tested | not tested | :heavy_check_mark: | :heavy_check_mark:  |

https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/62bf123fa2ccdf0009b2b861?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage]=staging

https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/63e2a043694c440009a3ec7c?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage]=staging

https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/63e2b01a5f6f3500084cc414?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage]=staging

### Conclusion
I am moving this to Ready To Go and if you agree we can merge, because the situation is improving. Before enabling the 'Background Reports' feature toggle in production we should still fix the remaining errors 500 - my opinion. I will open a follow up issue of this one.

**Pull request merged**